AsyncDisplayKit/ASDisplayNode.mm

```diff
@@ -814,6 +814,40 @@ - (void)setUsesImplicitHierarchyManagement:(BOOL)value
   _usesImplicitHierarchyManagement = value;
 }
 
+- (BOOL)_hasTransitionInProgress
+{
+    ASDN::MutexLocker l(_propertyLock);
+    return _transitionInProgress;
+}
+
+/// Starts a new transition and returns the transition id
+- (int32_t)_startNewTransition
+{
+  ASDN::MutexLocker l(_propertyLock);
+  _transitionInProgress = YES;
+    
+  if (!_transitionSentinel) {
+    _transitionSentinel = [[ASSentinel alloc] init];
+  }
```

Comment:
@nguyenhuy @appleguy What do you guys think of using an instance variable that we atomically increase for that instead of the sentinel object (who does basically the same but has the overhead of creating and destroying an object ?